clusterloader2/api/validation/validation.go

```diff
@@ -0,0 +1,180 @@
+	}
+	return allErrs
+}
+
+func verifyStep(s *api.Step, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i := range s.Phases {
+		allErrs = append(allErrs, verifyPhase(&s.Phases[i], fldPath.Child("phases").Index(i))...)
+	}
+	for i := range s.Measurements {
+		allErrs = append(allErrs, verifyMeasurement(&s.Measurements[i], fldPath.Child("measurements").Index(i))...)
+	}
+	isPhasesEmpty := len(s.Phases) == 0
+	isMeasurementsEmpty := len(s.Measurements) == 0
+	if isPhasesEmpty && isMeasurementsEmpty {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("measurements"), s.Measurements, "can't be empty with empty phases"))
+	}
+	if !isPhasesEmpty && !isMeasurementsEmpty {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("measurements"), s.Measurements, "can't be non-empty with non-empty phases"))
+	}
+	return allErrs
+}
+
+func verifyPhase(p *api.Phase, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if p.NamespaceRange != nil {
+		allErrs = append(allErrs, verifyNamespaceRange(p.NamespaceRange, fldPath.Child("namespaceRange"))...)
+	}
+	if p.ReplicasPerNamespace < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicasPerNamespace"), p.ReplicasPerNamespace, "must be non-negative"))
+	}
+	for i := range p.ObjectBundle {
+		allErrs = append(allErrs, verifyObject(&p.ObjectBundle[i], fldPath.Child("objectBundle").Index(i))...)
+	}
+	return allErrs
+}
+
+func verifyObject(o *api.Object, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if o.ObjectType.Kind == "Service" {
+		errs := validation.NameIsDNS1035Label(o.Basename, false)
+		for i := range errs {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("basename"), o.Basename, errs[i]))
+		}
+	} else {
+		errs := validation.NameIsDNSSubdomain(o.Basename, false)
+		for i := range errs {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("basename"), o.Basename, errs[i]))
+		}
+	}
+	if !fileExists(o.ObjectTemplatePath) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("objectTemplatePath"), o.ObjectTemplatePath, "file must exists"))
+	}
+	allErrs = append(allErrs, verifyObjectType(&o.ObjectType, fldPath.Child("objectType"))...)
+	return allErrs
+}
+
+func verifyNamespaceRange(nr *api.NamespaceRange, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if nr.Min < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("min"), nr.Min, "must be non-negative"))
+	}
+	if nr.Max < nr.Min {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("max"), nr.Max, "must be greater than min"))
+	}
+	return allErrs
+}
+
+func verifyObjectType(objectType *api.ObjectType, fldPath *field.Path) field.ErrorList {
+	fmt.Printf("all: %v\n", legacyscheme.Scheme.AllKnownTypes())
+	for t := range legacyscheme.Scheme.KnownTypes(schema.GroupVersion{Group: objectType.APIGroup, Version: objectType.APIVersion}) {
```

Comment:
See my comment here: https://github.com/kubernetes/perf-tests/pull/148/files#r200572077

I'm now pretty convinced that this is actually not needed and whole ObjectType should be removed.

Reply:
I'm removing object type verification.